fiftyone/core/map/process.py

```diff
@@ -0,0 +1,302 @@
+                or fou.recommend_process_pool_workers()
+            )
+
+        if config.max_process_pool_workers is not None:
+            workers = min(workers, config.max_process_pool_workers)
+
+        return cls(batch_cls, workers, batch_size)
+
+    def _map_samples_multiple_workers(
+        self,
+        sample_collection: SampleCollection[T],
+        map_fcn: Callable[[T], R],
+        *_,
+        progress: Union[bool, Literal["workers"], None],
+        save: bool,
+        skip_failures: bool,
+    ) -> Iterator[Tuple[bson.ObjectId, R]]:
+        ctx = fou.get_multiprocessing_context()
+
+        if progress is None:
+            progress = fo.config.show_progress_bars
+
+        if progress == "workers":
+            worker_progress = True
+            progress = False
+            lock = ctx.RLock()
+            tqdm.set_lock(lock)
+        else:
+            worker_progress = False
+            lock = None
+
+        queue = multiprocessing.Queue()
+        batch_count = multiprocessing.Value("i", 0)
+        sample_count = (
+            multiprocessing.Value("i", 0) if progress is not False else None
+        )
+        cancel_event = multiprocessing.Event()
+
+        # Extract information from sample collection.
+        if isinstance(sample_collection, fov.DatasetView):
+            # pylint:disable-next=protected-access
+            dataset_name = sample_collection._root_dataset.name
+            # pylint:disable-next=protected-access
+            view_stages = sample_collection._serialize()
+        else:
+            dataset_name = sample_collection.name
+            view_stages = None
+
+        sample_batches = self._batch_cls.split(
+            sample_collection, self.workers, self.batch_size
+        )
+
+        pool = ctx.Pool(
+            processes=self.workers,
+            initializer=_init_worker,
+            initargs=(
+                dataset_name,
+                view_stages,
+                pickle.dumps(map_fcn),
+                batch_count,
+                sample_count,
+                queue,
+                cancel_event,
+                save,
+                worker_progress,
+                skip_failures,
+                lock,
+            ),
+        )
+
+        pb = fou.ProgressBar(
+            total=sum(batch.total for batch in sample_batches),
+            progress=progress,
+        )
+
+        num_batches = len(sample_batches)
+        with pool, pb:
+            pool.map_async(
+                _map_batch,
+                (
+                    (idx + 1, num_batches, batch)
+                    for idx, batch in enumerate(sample_batches)
+                ),
+            )
+
+            sample_errors: List[Tuple[bson.ObjectId, Exception, None]] = []
+
+            # Initialize backoff parameters
+            initial_timeout = 0.1
+            max_timeout = 5.0
+            backoff_factor = 2
+            current_timeout = initial_timeout
```

Comment:
can we not utilize a back off library?

Reply:
I have opened a draft PR to utilize backoff library and will do more testing tomorrow; just checking, is this a blocking request?
https://github.com/voxel51/fiftyone/pull/5749

Reply:
no